modules/video/src/videoplayer.cpp

```diff
@@ -0,0 +1,949 @@
+    if (!isMaster) {
+        if ((_correctPlaybackTime - _currentVideoTime) > glm::epsilon<double>()) {
+            seekToTime(_correctPlaybackTime, isMappingTime);
+        }
+    }
+}
+
+const std::unique_ptr<ghoul::opengl::Texture>& VideoPlayer::frameTexture() const {
+    return _frameTexture;
+}
+
+void VideoPlayer::reset() {
+    if (_videoFile.empty()) {
+        return;
+    }
+    destroy();
+    _isDestroying = false;
+    initializeMpv();
+}
+
+bool VideoPlayer::isInitialized() const {
+    return _isInitialized;
+}
+
+bool VideoPlayer::isWithingStartEndTime() const {
+    const double now = global::timeManager->time().j2000Seconds();
+    return now <= _endJ200Time && now >= _startJ200Time;
+}
+
+void VideoPlayer::updateFrameDuration() {
+    double openspaceVideoLength = (_endJ200Time - _startJ200Time) / _videoDuration;
+    _frameDuration = (1.0 / _fps) * openspaceVideoLength;
+}
+
+double VideoPlayer::correctVideoPlaybackTime() const {
+    const double now = global::timeManager->time().j2000Seconds();
+    double percentage = 0.0;
+    if (now > _endJ200Time) {
+        percentage = 1.0;
+    }
+    else if (now < _startJ200Time) {
+        percentage = 0.0;
+    }
+    else {
+        percentage = (now - _startJ200Time) / (_endJ200Time - _startJ200Time);
+    }
+    return percentage * _videoDuration;
+}
+
+void VideoPlayer::createFBO(int width, int height) {
+    LINFO(fmt::format("Creating new FBO with width: {} and height: {}", width, height));
+
+    if (width <= 0 || height <= 0) {
+        LERROR("Cannot create empty fbo");
+        return;
+    }
+
+    // Update resolution of video
+    _videoResolution = glm::ivec2(width, height);
+
+    glGenFramebuffers(1, &_fbo);
+    glBindFramebuffer(GL_FRAMEBUFFER, _fbo);
+
+    _frameTexture = std::make_unique<ghoul::opengl::Texture>(
+        glm::uvec3(width, height, 1),
+        GL_TEXTURE_2D
+    );
+    _frameTexture->uploadTexture();
+
+    // Configure
+    _frameTexture->bind();
+    glPixelStorei(GL_PACK_ALIGNMENT, 1);
+    glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
+
+    // Disable mipmaps
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_BASE_LEVEL, 0);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAX_LEVEL, 0);
+
+    // Bind texture to framebuffer
+    glFramebufferTexture2D(
+        GL_FRAMEBUFFER,
+        GL_COLOR_ATTACHMENT0,
+        GL_TEXTURE_2D,
+        *_frameTexture,
+        0
+    );
+
+    // Unbind FBO
+    glBindFramebuffer(GL_FRAMEBUFFER, 0);
+}
+
+void VideoPlayer::resizeFBO(int width, int height) {
+    if (width == _videoResolution.x && height == _videoResolution.y) {
+        return;
+    }
+    LINFO(fmt::format("Resizing FBO with width: {} and height: {}", width, height));
+
+    // Update resolution of video
+    _videoResolution = glm::ivec2(width, height);
+
+    // Delete old FBO and texture
+    glDeleteFramebuffers(1, &_fbo);
+    _frameTexture.reset(nullptr);
+
+    createFBO(width, height);
+}
+
+VideoPlayer::~VideoPlayer() {}
```

Comment:
I'd put the destructor just after the constructor since they belong logically together